### PR TITLE
docs: Add a roadmap

### PR DESCRIPTION
**Pull request by Tarrasch**
## Description

Add a roadmap. A description of the roadmap is given inside the roadmap markdown file.

### CODE REVIEW

**Comment by Tarrasch** [comment]
Does this look sensible @ulzha? If so we could merge this and we could add #1897 into the roadmap. :)

**Comment by Tarrasch** [comment]
Founders @erikbern and @freider might want to review this as well.

**Comment by ulzha** [comment]
Great that you bring this up! It could be perceived as slightly too much set in stone though, by an accidental reader. Perhaps a bigger disclaimer at the top is needed, saying that the document is work in progress, more like an ideas dump so far?

I believe we should get the contributor community to participate in shaping the roadmap. Wonder what medium we could use to discuss and vote at a high enough level. E.g. previously there was at least a features [survey](https://groups.google.com/d/msg/luigi-user/0JIj0tRcjPA/w57wt03eSIkJ) in luigi-user@; we could do a similar one covering roadmap beyond features, but an email thread is perhaps not the medium I'm looking for...

And then the roadmap in document form should contain info how it was formed, i.e. how the selected issues qualified for the roadmap among _all_ the [issues](https://github.com/spotify/luigi/issues).

**Comment by Tarrasch** [comment]
Thanks for your feedback uldis. I'll change this to a slightly different format. Which I'll just do rather than explain and then you can review again. I'll ping you then!

I think the general theme for all of the items I put are that they are about decreasing the maintenance burden and increase general quality. But not so much about throwing in more features in luigi.

**Comment by mfcabrera** [comment]
Just my two cents here, I am luigi user looking for a way to contribute.  Maybe create some issues with some tags (i.e. "roadmap" and "easy"), so other people can contribute. an "easy" tag would tags for new comers that want to participate.

**Comment by Tarrasch** [comment]
@mfcabrera, thanks for your input. It's up and coming contributors like you I partially would like to target.

So basically your suggestion is to use GitHub issues instead of a committed files in a roadmap/ directory?

**Comment by mfcabrera** [comment]
@Tarrasch I think both would work fine. Specific issues will allow contributors to take ownership. (i.e. I will work on this) whereas the Roadmap will contain the big picture/ideas.

**Comment by Tarrasch** [comment]
Yea. I think instead of adding one more document (this) I should put effort into cleaning existing issues. There's many old issues and low quality issues that should just be closed, while others should be labeled.

I've deliberately written pretty small tasks in this roadmap. Hence it's maybe not the long-term roadmap one might expect. The reason I wrote small tasks is because I believe long-term goals never gets done anyway (unless someone is paid to work on it). So perhaps the content I've written here better fits as issues.

Thanks for your feedback @ulzha and @mfcabrera.

**Comment by Tarrasch** [comment]
Closing with the motivation I wrote above.

**Comment by dlstadther** [comment]
@Tarrasch Regarding your comment above, `spotify/luigi` currently has >200 open issues (as well as a log of orphaned PRs). Many of these are possible bugs or questions going as far back as mid 2013. How do you recommend labeling and/or closing these? i.e. issues older than a year get closed; no follow-up replies after X months gets closed; etc.

I imagine luigi is still a serious ETL tool option for many. So it makes sense that we should spend some time cleaning up the community. Thoughts?

File: doc/roadmap/README.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+For sanitizing console output and logging. Sounds great! I I really want to get rid of our own "luigi-interface" convention.
+Other priorities we have that are being worked on by DataEx (Uldis' squad) is sanitizing console output & logging, and removing the dependency on cron
+
+## Cron replacement
+
+Arash thinks this is a quite big task (in contrast to the other things in this
+roadmap). But in the very long run, this could become relevant.
+
+## Unify task namespaces with python namespaces
```
Comment (Tarrasch) [inline doc/roadmap/README.md]: This is resolved now as of #2000 :D

**Comment by Tarrasch** [comment]
You're totally right in that it makes sense to do some clean up work. I also believe it's worth while for a serious ETL tool like luigi.

Here's some of my thoughts and opinions:

 * We can make quick estimates of difficulty, Easy-Medium-Hard.
 * We can categorize tickets into core (luigi parameters, scheduler, visualizer, luigi workers, etc.) vs contrib (like Hadoop, AWS, sqlalchemy targets etc.).
 * I'm not sure if a "everything older than Y days should be closed" rule is good. There are some old ideas that actually are still valuable today.
   *  However, I think anything inactive for more than 4 months that's either "hard" or "contrib" can be closed, I don't think it's likely anyone will sporadically want to pick these up ever. Anyone can feel free to reopen the issues though.

This should make the issues easier to search. Also we can probably  close at least half of the current open issues. What do you think? If you agree to some of it. Maybe we should put this in a document?

**Comment by mfcabrera** [comment]
@Tarrasch I seen now some labels with more or less the same structure. Is there a way I can help here? with  the ticket categorization/clean-up? What is the channel of communications for Contributors/Collaborators?

**Comment by Tarrasch** [comment]
>  I seen now some labels with more or less the same structure.

Oh yes you're right! Well. We can start to apply a inactive for >4 month rule now then. I'll try to triage some more tickets now.

> Is there a way I can help here? with the ticket categorization/clean-up?

I think you can start in any part you feel passionate about. Like is there any part of luigi that has annoyed you?

> What is the channel of communications for Contributors/Collaborators?

There is no dedicated channel. Maybe the users list can do for a start, once we think it's worth dividing the mail list in users vs contributors we can maybe decide on that when it becomes an issue. Does that sound reasonable?

**Comment by mfcabrera** [comment]
Hi @Tarrasch!

I don't have any particular issue (but I've been using Luigi for 3 years already) so I am familiar with the structure. I can also help with coding in the big changes if necessary, 

In order to help with the categorization clean-up I think I need to be made collaborator first. I cannot add any labels to issues right now.

**Comment by Tarrasch** [comment]
@mfcabrera I just added you  as a collaborator.

I recently started to [close a lot of old issues (started from back)](https://github.com/spotify/luigi/issues?q=is%3Aissue+is%3Aclosed+sort%3Aupdated-desc). Do you think I do it in a polite and good way?

**Comment by dlstadther** [comment]
@Tarrasch I think your comments for closing those issues was clear and polite. As i have time in the evenings, I'll help clean.

**Comment by mfcabrera** [comment]
@Tarrasch I also think the comments are clear and polite. If after one year you haven't update a PR you probably won't do it anyways.

**Comment by mfcabrera** [comment]
Just an idea.

Maybe it would be good to have a label for all things Web UI? (or even for some sensible parts of the core).

**Comment by Tarrasch** [comment]
> Maybe it would be good to have a label for all things Web UI? (or even for some sensible parts of the core).

Sure, `core` could be split into clients and luigid-related code. Feel free to add a label. :)

**Comment by dckc** [comment]
I'm interested to help with issue triage as well.

**Pull request closed without merging**